### PR TITLE
Update: allow disabling new-cap on object methods (fixes #3172)

### LINKED ISSUE #3172: new-cap on object method
I have a function constructor stored on an object that I would create like this:

```
var fooComponent = new obj.component();
```

I don't want obj to need be capitalized since the constructor is . JSHint's `newcap` behavior does this and is what I want.

```
  62:33  error  A constructor name should not start with a lowercase letter  new-cap
```

### PR DESCRIPTION
**Pull request by BYK**

### CODE REVIEW

File: tests/lib/rules/new-cap.js [hunk 1]
```diff
@@ -59,7 +59,8 @@ ruleTester.run("new-cap", rule, {
         { code: "var x = Foo.Bar(42);", options: [{ capIsNewExceptions: ["Bar"] }] },
         { code: "var x = Foo.Bar(42);", options: [{ capIsNewExceptions: ["Foo.Bar"] }] },
         { code: "var x = new foo.bar(42);", options: [{ newIsCapExceptions: ["bar"] }] },
-        { code: "var x = new foo.bar(42);", options: [{ newIsCapExceptions: ["foo.bar"] }] }
+        { code: "var x = new foo.bar(42);", options: [{ newIsCapExceptions: ["foo.bar"] }] },
```
Comment (ilyavolodin) [inline tests/lib/rules/new-cap.js]: Can you add a test with `properties` and `capIsNew`? `isCapAllowed` that you modified is called for both, and I want to make sure we have tests for that.

Comment (BYK) [inline tests/lib/rules/new-cap.js]: Upcoming, good catch!

**Comment by ilyavolodin** [comment]
@eslint/eslint-team @BYK added some additional tests, but I'm having trouble figuring out if the following use case should pass:

```js
{ code: "var x = Foo.bar(42);", options: [{ capIsNew: true, properties: false }] }
```

The only use-case I can come up with for that is a "static" method on an object. Any opinions?

**Comment by gyandeeps** [comment]
I think that should pass as right now in `eslint` we have `CLIEngine` which has 2 static methods attached to it. So whenever someone uses that then this rule will error on them.

**Comment by nzakas** [comment]
Agree with @gyandeeps, I think it should pass. 

@BYK good to go?

**Comment by BYK** [comment]
>@BYK good to go?

I'm good and that should be the existing version in tests. Unless we need a rebase, patch should be ready to merge.

**Pull request merged**